### DISCUSSION TITLE
Connection Request issues when running "ls"

### DISCUSSION THREAD

**Discussion by MrSpock** [guillaumepellegrino/acsrs#19] [Q&A]
Hi,

I'm evaluating this acs implementation but I stuck with some commands on ZTE devices.
When executing "ls" a see following logs in verbose mode:

```
13:21:40 [INFO] Send ConnectionRequest to http://10.40.15.63:58000
13:21:40 [INFO] ConnectionRequest was acknowledged
13:21:40 [INFO] [SN:??][SID:??][1] Authorization required
```

And after some timeout cli shows message:
```
Error: "Failed to transfer request to CPE: ACS server failed to get a response from CPE\n\nCaused by:\n    deadline has elapsed\n\nLocation:\n    /home/spock/.cargo/registry/src/in
dex.crates.io-6f17d22bba15001f/acsrs-0.5.1/src/mng.rs:91:14"
```
After some time additional lines comes up:
```
[ERROR] Failed to serve connection: hyper::Error(Shutdown, Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" })
```
I see in config file that username/password pair matches the one seen on device webpage, however I'm getting such error quite often.

after few 'ls' commands I'm getting response with following logs:
```
13:29:55 [INFO] [SN:ZTEEQERK9A12710][SID:35][4] Send: Transfer pending GetParameterNames to CPE
13:29:55 [INFO] [SN:ZTEEQERK9A12710][SID:35][5] Received: GPN Response
```
Any idea ?

Thanks!

**Comment by guillaumepellegrino** [comment]
Hello @MrSpock ,
It looks like to me the authentication procedure does not go up to the end. Hard to say why.
In case of an invalid password, I would expect an "Access forbidden" log.

Maybe a wireshark capture could help ?
